push-messaging-and-notifications/main.js

```diff
@@ -76,7 +94,7 @@ function subscribe() {
   pushButton.disabled = true;
 
   navigator.serviceWorker.ready.then(function(serviceWorkerRegistration) {
-    serviceWorkerRegistration.pushManager.subscribe()
+    serviceWorkerRegistration.pushManager.subscribe({userVisibleOnly: true})
       .then(function(subscription) {
         // The subscription was successful
```

Comment:
I would move the concatenation logic to here, and then you'll have the full string ready to pass to `showCurlCommand()` and `sendSubscriptionToServer()`